src/backend/distributed/deparser/deparse_database_stmts.c

```diff
@@ -76,37 +76,19 @@ AppendGrantOnDatabaseStmt(StringInfo buf, GrantStmt *stmt)
 
 	appendStringInfo(buf, "%s ", stmt->is_grant ? "GRANT" : "REVOKE");
 
-	if (!stmt->is_grant && stmt->grant_option)
-	{
-		appendStringInfo(buf, "GRANT OPTION FOR ");
-	}
+	AppendGrantOptionFor(buf, stmt);
 
 	AppendGrantPrivileges(buf, stmt);
 
 	AppendGrantDatabases(buf, stmt);
 
 	AppendGrantGrantees(buf, stmt);
 
-	if (stmt->is_grant && stmt->grant_option)
-	{
-		appendStringInfo(buf, " WITH GRANT OPTION");
-	}
-	if (!stmt->is_grant)
-	{
-		if (stmt->behavior == DROP_RESTRICT)
-		{
-			appendStringInfo(buf, " RESTRICT");
-		}
-		else if (stmt->behavior == DROP_CASCADE)
-		{
-			appendStringInfo(buf, " CASCADE");
-		}
-	}
+	AppendWithGrantOption(buf, stmt);
 
-	if (stmt->grantor)
-	{
-		appendStringInfo(buf, " GRANTED BY %s", RoleSpecString(stmt->grantor, true));
-	}
+	AppendGrantRestrictAndCascade(buf, stmt);
+
+	AppendGrantedByInGrant(buf, stmt);
```

Comment:
This line should be added to all of the other grants too. I think it might make sense to combine `AppendWithGrantOption` + `AppendGrantRestrictAndCascade` + `AppendGrantedByInGrant` into a single function. Since it's a shared suffix for all grants. That function could be called something like `AppendGrantSharedSuffix`.

Reply:
And we can do the same for the shared prefix that all the grants have:
```
	appendStringInfo(buf, "%s ", stmt->is_grant ? "GRANT" : "REVOKE");

	AppendGrantOptionFor(buf, stmt);

	AppendGrantPrivileges(buf, stmt);
```

Reply:
Then the `AppendGrantOnDatabaseStmt` body would look like: 
```c
AppendGrantSharedPrefix(buf, stmt);
AppendGrantDatabase(buf, stmt);
AppendGrantSharedSuffix(buf, stmt);
```

Reply:
@JelteF fixed all the code places other than 'Create' role since this one has a different structure

Reply:
@JelteF Fixed your comments other than the AppendRole